packages/framer-motion/src/animation/utils/transitions.ts

```diff
@@ -277,18 +277,19 @@ export function startAnimation(
             transition,
             onComplete
         )
-        const delay = getDelayFromTransition(transition, key)
+        const delayBy = getDelayFromTransition(transition, key)
 
         const start = () => (controls = animation())
 
-        if (delay) {
-            delayTimer = window.setTimeout(start, secondsToMilliseconds(delay))
+        let cancelDelay: VoidFunction
+        if (delayBy) {
+            cancelDelay = delay(start, secondsToMilliseconds(delayBy))
         } else {
             start()
         }
 
         return () => {
-            clearTimeout(delayTimer)
+            cancelDelay && cancelDelay()
```

Comment:
Would it be safe to set `cancelDelay` to undefined after calling it?

Reply:
I don't think it's necessary, same as cancelTimeout etc nothing will happen if you keep calling it